src/narmvm.rs

```diff
@@ -552,6 +552,24 @@ impl NarmVM{
                     }
                     return Ok(0);
                 },
+                //0100_0110_xyyy_yzzz MOV reg T1(???) noflags (This op allows moving data to/from/between high regs)
+                0b0100_0110_0000_0000 => {
+                    
+                    if reg1.register == 15 && reg2.register == 15 {
+                        // "Use of r15 as a source register is deprecated when r15 is the destination register"
+                    }
+                    else if reg1.register == 15 {
+                        self.set_reg(&reg2, self.get_reg(&reg1)); // Dest get value of PC with last 2 bits zeroed, is that correct?
```

Comment:
PC should be written with the bottom bit cleared (technically bottom bit set would be an error, but we can ignore it) while PC should be read with bottom two bits cleared. See also https://developer.arm.com/documentation/ddi0419/c/Application-Level-Architecture/The-Thumb-Instruction-Set-Encoding/Thumb-instruction-set-encoding/Use-of-0b1111-as-a-register-specifier?lang=en

Reply:
I believe in this context the `virtual_pc` variable should actually be used rather than get_reg because of this (virtual_pc is basically current_instruction_location + 4)

Reply:
I hadn't found that specification, will do my best to align with it.

Reply:
Done, I think